suite-native/module-home/src/components/EmptyConnectedDeviceState.tsx

```diff
@@ -11,6 +12,7 @@ const cardStyle = prepareNativeStyle(utils => ({
 }));
 
 export const EmptyConnectedDeviceState = () => {
+    useDeviceConnect();
```

Comment:
Is it OK to use this hook in multiple components? What will happen if these components will be mounted same time? cc @juriczech

Reply:
We're already discussing with @vytick. It was meant to be used universally. However right now, when used in homescreen, there is cyclic deps because of the condition which redirects to HODL screen.

```
       if (isDeviceConnectedAndAuthorized && !isSelectedDeviceImported) {
            navigation.navigate(RootStackRoutes.ConnectDevice, {
                screen: ConnectDeviceStackRoutes.ConnectingDevice,
            });
        }
```

This is fulfilled every time on homescreen if device is connected and authorized, which redirects to loader screen, which redirects to homescreen. 

AFAIK @vytick is fixing this right now, but yes, should be working on multiple screens. 

By design, only redux selectors are used there so we can have multiple instances without clashes, altough I would prefer to use it in device manager (won't be flexible with this one https://github.com/trezor/trezor-suite/issues/9856) or in `AppTabsNavigator` maybe? Altough this is also fine, just when we implement disconnect within the hook, we will need to propagate this hook to all screens...

Reply:
@vytick Can you please try and place it to AppTabsNavigator? Shouldn't change behaviour on this screen, just the hooks functions will be reused in all of dashboards tabs and subscreens. If that's working, it would be more flexible IMO

Reply:
it still rebuilds over and over. I had to move it one level up to RootStackNavigator to stop rebuilding the tree

Reply:
@Nodonisko @vytick Actually it cannot be in the RootStack - this would enable user to skip onboarding. This was requested by @mnuky that we will not enable user to connect device in onboarding (or he can connect but won't be redirected to dashboard). This change would allow this to work so it needs to be level down (dashboard, eg app tabs), or we need to find better solution.

Reply:
My question is though - do you need this in Homescreen at all? Since the hook is used in Empty state as well as connect module, shouldn't the invalid firmware effect get triggered there? 🤔

Reply:
b17e3a181

Reply:
Before i added it to homescreen previously, it was not triggered while you changed device via device-manager. Lets talk it tmrw

Reply:
c4c7f8c

Reply:
I'm not sure about this check. Every new effect will need this check and seems like we're abusing it here a little bit. If @Nodonisko feels like it's fine, feel free to resolve this conversation, I don't want to hold this up too much.